stacker/tests/test_config.py

```diff
@@ -537,6 +537,20 @@ def test_raise_construct_error_on_duplicate_stack_name_dict(self):
         with self.assertRaises(ConstructorError):
             parse(yaml_config)
 
+    def test_parse_invalid_inner_keys(self):
```

Comment:
Do we already have a test that has excess keys at the top level that would cover that use case?

Reply:
`test_parse_with_arbitrary_anchors` seems to do the job: https://github.com/Cobliteam/stacker/blob/94cc1e8a4eea1eb7ee8329caa4b25a61cd98c648/stacker/tests/test_config.py#L168

Reply:
Awesome, thanks!